File: project_euler/01/010.py
#!/usr/bin/env python3
'''Copyright [2017] <dmnsn7@gmail.com>'''

N = 2000000


def prime_sieve(num_n):
    prime = []
    is_prime = [True for i in range(num_n)]
    for i in range(2, num_n):
        if is_prime[i]:
            prime.append(i)
        for _, pri in enumerate(prime):
            if pri * i >= num_n:
                break
            is_prime[pri * i] = False
            if i % pri == 0:
                break

    return prime


def main():
    prime = prime_sieve(N)
    add_up = 0
    for _, pri in enumerate(prime):
        add_up += pri

    print(add_up)


if __name__ == '__main__':
    main()

File: project_euler/01/010.go
// Copyright [2017] <dmnsn7@gmail.com>

package main

import "fmt"

const n = 2000000

func primeSieve(n int) []int {
	var prime []int
	isPrime := make([]bool, n)
	for i := 0; i < n; i++ {
		isPrime[i] = true
	}
	for i := 2; i < n; i++ {
		if isPrime[i] == true {
			prime = append(prime, i)
		}
		for j := 0; j < len(prime) && prime[j]*i < n; j++ {
			isPrime[prime[j]*i] = false
			if i%prime[j] == 0 {
				break
			}
		}
	}

	return prime
}

func main() {
	prime := primeSieve(n)
	sum := int64(0)
	for i := 0; i < len(prime); i++ {
		sum += int64(prime[i])
	}

	fmt.Printf("%d\n", sum)
}
